src/main/java/nwoolcan/model/utils/QuantityChecker.java

```diff
@@ -23,14 +23,14 @@ private QuantityChecker() { }
 
     /**
      * Performs a check on the {@link Quantity} passed by parameter, returning a {@link Result}
-     * with a {@link IllegalArgumentException} error if the quantity value is null or the quantity
```

Comment:
I think you should check also that the value is not null, is will be as easy as using `Results.requireNonNull(quantity)` instead of `Result.of(quantity)`